### PR TITLE
Add an entry on the right-click menu to add to top of build queue.

### PR DESCRIPTION
**Pull request by wheals**
As opposed to the bottom; suggested by @MatGB in #336.

### CODE REVIEW

File: UI/BuildDesignatorWnd.cpp [hunk 1]
```diff
@@ -438,6 +438,8 @@ class BuildDesignatorWnd::BuildSelector : public CUIWnd {
       * filter settings. */
     void    PopulateList();
 
+    void AddBuildItemToQueue(GG::ListBox::iterator it, bool top);
```
Comment (geoffthemedio) [inline UI/BuildDesignatorWnd.cpp]: indent to match

**Comment by geoffthemedio** [comment]
Rather than a "top" bool, I think it'd be better to make RequestNamedBuildItemSignal take an int parameter like TechTreeWnd::QueueAddTechsSignalType. BuildDesignatorWnd::BuildItemRequested and ProductionWnd::AddBuildToQueueSlot would then also take an int, like ResearchWnd::AddTechsToQueueSlot does, to specify the queue position.

**Comment by wheals** [comment]
The first commit there isn't strictly necessary, but I figured I might as well lower the number of functions I had to change to take an int instead of a bool.

**Comment by geoffthemedio** [comment]
Would be good if ctrl + double-click put stuff to the top of the queue, like happens on the research screen.

**Comment by wheals** [comment]
Should I open a new PR for that?

**Comment by wheals** [comment]
Or it should be easy to do it yourself -- looks to me like replacing line 1244 of BuildDesignatorWnd.cpp with `    AddBuildItemToQueue(it, modkeys & GG::MOD_KEY_CTRL);` should do the trick.

**Comment by wheals** [comment]
Oops, realised the solution I have doesn't work after all. I'll open a new PR for this.

**Pull request merged**